flytekit/extras/tasks/shell.py

```diff
@@ -109,6 +110,8 @@ def __init__(
         task_config: T = None,
         inputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
         output_locs: typing.Optional[typing.List[OutputLocation]] = None,
+        env: typing.Optional[typing.List[str]] = None,
```

Comment:
Will BOLO below, I would have expected this to be `typing.Optional[typing.Dict[str, typing.Type]]`

Reply:
(i.e., because env variables are named)

Reply:
Ah okay I see, this is just the list of names of variables, which are then extracted from kwargs

Reply:
ya, adding additional typing info here felt redundant and opened room for mismatch